src/set_theory/zfc/ordinal.lean

```diff
@@ -16,59 +16,127 @@ Further development can be found on the branch `von_neumann_v2`.
+local attribute [simp] subtype.coe_inj
 
 namespace Set
 
+/-! ### Transitive sets -/
+
 /-- A transitive set is one where every element is a subset. -/
-def is_transitive (x : Set) : Prop := ∀ y ∈ x, y ⊆ x
+def transitive (x : Set) : Prop := ∀ y ∈ x, y ⊆ x
 
-@[simp] theorem empty_is_transitive : is_transitive ∅ := λ y hy, (not_mem_empty y hy).elim
+@[simp] theorem empty_transitive : transitive ∅ := λ y hy, (not_mem_empty y hy).elim
 
-theorem is_transitive.subset_of_mem (h : x.is_transitive) : y ∈ x → y ⊆ x := h y
+theorem transitive.subset_of_mem (h : x.transitive) : y ∈ x → y ⊆ x := h y
 
-theorem is_transitive_iff_mem_trans : z.is_transitive ↔ ∀ {x y : Set}, x ∈ y → y ∈ z → x ∈ z :=
+theorem transitive_iff_mem_trans : z.transitive ↔ ∀ ⦃x y : Set⦄, x ∈ y → y ∈ z → x ∈ z :=
 ⟨λ h x y hx hy, h.subset_of_mem hy hx, λ H x hx y hy, H hy hx⟩
 
-alias is_transitive_iff_mem_trans ↔ is_transitive.mem_trans _
+theorem transitive.mem_trans (h : z.transitive) : ∀ {x y : Set}, x ∈ y → y ∈ z → x ∈ z :=
+transitive_iff_mem_trans.mp h
 
-protected theorem is_transitive.inter (hx : x.is_transitive) (hy : y.is_transitive) :
-  (x ∩ y).is_transitive :=
+protected theorem transitive.inter (hx : x.transitive) (hy : y.transitive) :
+  (x ∩ y).transitive :=
 λ z hz w hw, by { rw mem_inter at hz ⊢, exact ⟨hx.mem_trans hw hz.1, hy.mem_trans hw hz.2⟩ }
 
-protected theorem is_transitive.sUnion (h : x.is_transitive) : (⋃₀ x).is_transitive :=
+protected theorem transitive.sUnion (h : x.transitive) : (⋃₀ x).transitive :=
 λ y hy z hz, begin
   rcases mem_sUnion.1 hy with ⟨w, hw, hw'⟩,
   exact mem_sUnion_of_mem hz (h.mem_trans hw' hw)
 end
 
-theorem is_transitive.sUnion' (H : ∀ y ∈ x, is_transitive y) : (⋃₀ x).is_transitive :=
+theorem transitive.sUnion' (H : ∀ y ∈ x, transitive y) : (⋃₀ x).transitive :=
 λ y hy z hz, begin
   rcases mem_sUnion.1 hy with ⟨w, hw, hw'⟩,
   exact mem_sUnion_of_mem ((H w hw).mem_trans hz hw') hw
 end
 
-theorem is_transitive_iff_sUnion_subset : x.is_transitive ↔ ⋃₀ x ⊆ x :=
+theorem transitive_iff_sUnion_subset : x.transitive ↔ ⋃₀ x ⊆ x :=
 ⟨λ h y hy, by { rcases mem_sUnion.1 hy with ⟨z, hz, hz'⟩, exact h.mem_trans hz' hz },
   λ H y hy z hz, H $ mem_sUnion_of_mem hz hy⟩
 
-alias is_transitive_iff_sUnion_subset ↔ is_transitive.sUnion_subset _
+alias transitive_iff_sUnion_subset ↔ transitive.sUnion_subset _
 
-theorem is_transitive_iff_subset_powerset : x.is_transitive ↔ x ⊆ powerset x :=
+theorem transitive_iff_subset_powerset : x.transitive ↔ x ⊆ powerset x :=
 ⟨λ h y hy, mem_powerset.2 $ h.subset_of_mem hy, λ H y hy z hz, mem_powerset.1 (H hy) hz⟩
 
-alias is_transitive_iff_subset_powerset ↔ is_transitive.subset_powerset _
+alias transitive_iff_subset_powerset ↔ transitive.subset_powerset _
+
+/-! ### Ordinals as sets -/
+
+/-- A set `x` is a von Neumann ordinal when it's a hereditarily transitive set. We prove that this
+further implies that `x` is well-ordered under `∈`. -/
+def is_ordinal (x : Set) : Prop := x.hereditarily transitive
```

Comment:
This is a really cool definition, much less ad-hoc than what I had come up with. That said, it would be great if we could prove the equivalence between definitions. Mine is like the RHS of `Set.is_ordinal_iff_is_trans`, but with fewer conditions.

Also, if we're going to use `transitive` instead of `is_transitive`, why not use `ordinal` instead of `is_ordinal`? Not sure if I would prefer that, but I'm interested in your reasoning.

Reply:
`transitive` is clearly a predicate, while `ordinal` is not. It may also cause some name conflicts, so I kept the name `is_ordinal`.

Reply:
Let's not rename this for now.